dev/TeachingTip/TeachingTip.cpp

```diff
@@ -991,23 +1045,36 @@ void TeachingTip::SetViewportChangedEvent()
 void TeachingTip::RevokeViewportChangedEvent()
 {
     m_targetEffectiveViewportChangedRevoker.revoke();
+    m_effectiveViewportChangedRevoker.revoke();
     m_targetLayoutUpdatedRevoker.revoke();
 }
 
 void TeachingTip::TargetLayoutUpdated(const winrt::IInspectable&, const winrt::IInspectable&)
 {
-    if (IsOpen() && m_target)
+    if (IsOpen())
     {
-        auto newTargetBounds = m_target.get().TransformToVisual(nullptr).TransformBounds({
-            0.0,
-            0.0,
-            static_cast<float>(m_target.get().as<winrt::FrameworkElement>().ActualWidth()),
-            static_cast<float>(m_target.get().as<winrt::FrameworkElement>().ActualHeight())
-            });
-        if (newTargetBounds != m_currentTargetBounds)
+        if (auto&& target = m_target.get())
         {
-            m_currentTargetBounds = newTargetBounds;
-            PositionPopup();
+            auto newTargetBounds = target.TransformToVisual(nullptr).TransformBounds({
+                0.0,
+                0.0,
+                static_cast<float>(target.as<winrt::FrameworkElement>().ActualWidth()),
```

Comment:
>target.as<winrt::FrameworkElement>() [](start = 35, length = 36)

as isn't free, so let's try not to do it multiple times. Also why is m_target typed as UIElement if we always need it as FrameworkElement?

Reply:
Good point, I'll change m_target to be a Framework element. For now I'm going to leave the SetAttach method as UIElement since I'll be removing it shortly in favor of the .Target property, which I'll update to be a Framework Element.